src/components/Button/Button.tsx

```diff
@@ -41,31 +42,50 @@ export const Button = ({
         size="sm"
       />
     )}
-    {label ? label : children}
+    <span data-testid="click-ui-button-label">{label ? label : children}</span>
     {iconRight && (
       <ButtonIcon
         name={iconRight}
         size="sm"
       />
     )}
+    {loading && (
+      <LoadinIconWrapper>
+        <Icon
+          name="loading-animated"
+          data-testid="click-ui-loading-icon"
+        />
+      </LoadinIconWrapper>
+    )}
   </StyledButton>
 );
 
+const LoadinIconWrapper = styled.div`
+  position: absolute;
```

Comment:
Why are we using absolute rendering ?
We can add a conditional to render the children

Reply:
if we do that, then the width of the button will change. In this way it will stay the same. We are basically painting a new layer on top that hides the content below